lms/djangoapps/teams/tests/test_views.py

```diff
@@ -792,6 +802,59 @@ def test_member_trying_to_get_private_topic(self):
         self.assertEqual('private_topic_1_id', result['results'][0]['topic_id'])
         self.assertNotEqual([], result['results'])
 
+    @ddt.unpack
+    @ddt.data(
+        ('student_masters_not_on_team', 1),
+        ('student_masters', 1),
+        ('student_enrolled', 0),
+        ('staff', 1),
+    )
+    def test_text_search_organization_protected(self, user, expected_results):
+        """
+        When doing a text search as different users, will the masters_only team show up?
+        Only staff, or people who are within the organization_protected bubble should be
+        able to see the masters team
+        """
+        self.reset_search_index()
+        result = self.get_teams_list(
+            data={'text_search': 'master'},
+            user=user,
+        )
+        self.assertEqual(result['count'], expected_results)
+
+    @ddt.unpack
+    @ddt.data(
+        ('student_on_team_1_private_set_1', True, True, True),
+        ('student_on_team_2_private_set_1', True, True, True),
+        ('student_enrolled', False, False, False),
+        ('student_masters', True, True, True),
+        ('staff', True, True, True),
+    )
+    def test_text_search_private_teamset(self, user, can_see_private_1_1, can_see_private_1_2, can_see_private_2_1):
+        """
+        When doing a text search as different users, will private_managed teams show up?
+        Only staff should be able to see all private_managed teams.
+        Students enrolled in a private_managed teams should be able to see their team, and no others.
+
+        TODO: Currently, anyone within the organization_protected can see all private teams regardless of
```

Comment:
Nit: I think you're missing the word "teamset" after "organization_protected"?

Reply:
It's actually 'bubble'